VOCS offset set and description read fixes

Thalley opened the pull request:
Fixes an issue with setting the VOCS offset and another issue when reading the description.

Thalley committed:
Bluetooth: Audio: Fix issues with setting VOCS offset state

Fixes 2 issues:
1) The write callback always returned an error
2) The auto-read of the offset state did not, unlike what the
comment suggested, retry only once and fail on the second attempt,
but rather keep retrying.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Remove VOCS desc data check as it can be empty

If the output description of the server is empty (legal by the
spec), then we should not return an error to the application
when read. Also added some debug statements when the value is
truncated.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

joerchan commented on subsys/bluetooth/audio/vocs_client.c:
```diff
@@ -300,9 +300,6 @@ static uint8_t vcs_client_read_output_desc_cb(struct bt_conn *conn, uint8_t err,
 		BT_DBG("Output description: %s", log_strdup(desc));
-	} else {
-		BT_DBG("Invalid output");
-		cb_err = BT_ATT_ERR_UNLIKELY;
```
You could remove cb_err now, since you only assign it and pass it forward.

Thalley replied:
Re-added the cb_err as we need to check for bad reads as well.

joerchan commented on subsys/bluetooth/audio/vocs_client.c:
```diff
@@ -230,6 +230,7 @@ static void vcs_client_write_vocs_cp_cb(struct bt_conn *conn, uint8_t err,
 {
 	int cb_err = err;
 	struct bt_vocs *inst = lookup_vocs_by_handle(conn, params->handle);
+	static bool retry = true;
```
Global for all connections? Shouldn't this be in the vocs instance?

Thalley replied:
Great catch! I had almost forgot I added multi-conn support :D

Thalley committed:
Bluetooth: Audio: VOCS add error checks on read

The error check was removed earlier as it was assumed that
we would either get an error, or the data would be valid.
However, without an error check, we are not guarded
against bad reads.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

asbjornsabo commented on subsys/bluetooth/audio/vocs_client.c:
```diff
@@ -290,18 +295,28 @@ static uint8_t vcs_client_read_output_desc_cb(struct bt_conn *conn, uint8_t err,
+		}
+		return BT_GATT_ITER_STOP;
+	}
+
```
I think it is possible to remove the duplicated call to the callback, if one wants.

```
if (cb_err) {
        BT_DBG
} else if (data) {
        /* Handle the data */
}
[...]
if (inst.cli [...] ) {
        inst->cli.cb->description(conn, inst, (cb_err ? NULL, desc))
}

```

Thalley replied:
It would be, but then I'd have to move lines 322-335 into an else clause to avoid setting the NULl terminator and attempting to print an invalid output description (or use a `goto` which I'm generally not a fan of).

Thalley replied:
Ended up doing it anyhow :)

asbjornsabo commented on subsys/bluetooth/audio/vocs_client.c:
```diff
@@ -119,6 +119,14 @@ static uint8_t vocs_client_read_offset_state_cb(struct bt_conn *conn, uint8_t er
+		}
+		return BT_GATT_ITER_STOP;
+	}
+
```
Possible to join callbacks?

Thalley replied:
This was a pretty easy one to join

asbjornsabo commented on subsys/bluetooth/audio/vocs_client.c:
```diff
@@ -157,6 +165,14 @@ static uint8_t vocs_client_read_location_cb(struct bt_conn *conn, uint8_t err,
+		}
+		return BT_GATT_ITER_STOP;
+	}
+
```
Possible to join callbacks?

Thalley replied:
This was a pretty easy one to join

asbjornsabo reviewed (commented):
This seems to work.  I have some comments to consider, though.

Thalley commented on subsys/bluetooth/audio/vocs_client.c:
```diff
@@ -210,7 +210,7 @@ static uint8_t internal_read_volume_offset_state_cb(struct bt_conn *conn, uint8_
 		}
 	} else {
-		BT_DBG("Invalid location");
+		BT_DBG("Invalid (empty) offset state read");
```
Do you mean line 212? :s

asbjornsabo replied:
I probably meant what was line 191, and which is now 202.
And I see that has been fixed.

asbjornsabo reviewed (approved):
My comments have been resolved to my satisfaction.
